### ISSUE TITLE
How many epochs should be trained?

### ISSUE THREAD

**Issue by xzhang2523** [Bigpig4396/PyTorch-Constrained-Policy-Optimization-CPO#1]
Hi,

I run you code for 1000 epochs, 
epi 986 reward 0.0 cost 0.11422845691382766 MC 499
epi 987 reward 0.0 cost 0.10420841683366733 MC 499
epi 988 reward 0.0 cost 0.1523046092184369 MC 499
epi 989 reward 0.0 cost 0.09418837675350701 MC 499
epi 990 reward 0.0 cost 0.10020040080160321 MC 499
epi 991 reward 0.0 cost 0.11422845691382766 MC 499
epi 992 reward 0.0 cost 0.056112224448897796 MC 499
epi 993 reward 0.0 cost 0.07414829659318638 MC 499
epi 994 reward 0.0 cost 0.05811623246492986 MC 499
epi 995 reward 0.0 cost 0.06212424849699399 MC 499
epi 996 reward 0.0 cost 0.09218436873747494 MC 499
epi 997 reward 0.0 cost 0.09619238476953908 MC 499
epi 998 reward 0.0 cost 0.11623246492985972 MC 499
epi 999 reward 0.0 cost 0.0781563126252505 MC 499

The reward is still 0, should I increase the epoch?

**Comment by Bigpig4396** [comment]
It can be possible that sparse reward can hardly be sampled in this particular environment, try to re-run it multiple times.

**Comment by xzhang2523** [comment]
Thanks for you kind reply.
Have you tried your code on the safeGym envs like pointNav, AntNav?

**Comment by xzhang2523** [comment]
I am curious that, do you have a wechat ID. 
Maybe we can exchange some information via this famous communication tool.
